src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp

```diff
@@ -0,0 +1,164 @@
+ */
+class GaugeWave : public MarkAsAnalyticSolution {
+ public:
+  static constexpr size_t volume_dim = 3;
+  struct Amplitude {
+    using type = double;
+    static constexpr OptionString help = {"Amplitude of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+  struct Wavelength {
+    using type = double;
+    static constexpr OptionString help = {"Wavelength of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  using options = tmpl::list<Amplitude, Wavelength>;
+  static constexpr OptionString help{"Gauge wave in flat spacetime"};
+
+  GaugeWave(double amplitude, double wavelength,
+            const OptionContext& context = {});
+
+  explicit GaugeWave(CkMigrateMessage* /*unused*/) noexcept {}
+
+  GaugeWave() = default;
+  GaugeWave(const GaugeWave& /*rhs*/) = default;
+  GaugeWave& operator=(const GaugeWave& /*rhs*/) = default;
+  GaugeWave(GaugeWave&& /*rhs*/) noexcept = default;
+  GaugeWave& operator=(GaugeWave&& /*rhs*/) noexcept = default;
+  ~GaugeWave() = default;
+
+  template <typename DataType>
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataType>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using tags = tmpl::list<
+      gr::Tags::Lapse<DataType>, ::Tags::dt<gr::Tags::Lapse<DataType>>,
+      DerivLapse<DataType>,
+      gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>>,
+      DerivShift<DataType>,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>>,
+      DerivSpatialMetric<DataType>, gr::Tags::SqrtDetSpatialMetric<DataType>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataType>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataType>>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<tags<DataType>> variables(
+      const tnsr::I<DataType, volume_dim>& x, double t,
+      tags<DataType> /*meta*/) const noexcept;
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  SPECTRE_ALWAYS_INLINE double amplitude() const noexcept { return amplitude_; }
+  SPECTRE_ALWAYS_INLINE double wavelength() const noexcept {
+    return wavelength_;
+  }
+
+ private:
+  double amplitude_{1.0}, wavelength_{1.0};
```

Comment:
I think these should be `signaling_NaN` and you set the defaults in the constructor (not sure though).